Type annotation for parametrize ids should allow HIDDEN_PARAM

Current the `ids` keyword argument of `pytest.mark.parametrize` is an `Iterable[None | str | float | int | bool]`. It should also allow `pytest.HIDDEN_PARAM` (similar to `pytest.param(id=...)`).

```console
$ pip list
Package           Version                 Editable project location
----------------- ----------------------- -------------------------
iniconfig         2.3.0
librt             0.8.1
mypy              1.19.1
mypy_extensions   1.1.0
packaging         26.0
pathspec          1.0.4
pip               24.0
pluggy            1.6.0
Pygments          2.19.2
pytest            9.1.0.dev216+g9fd1eebd8 /home/user/pytest
typing_extensions 4.15.0

$ pytest --version
pytest 9.1.0.dev216+g9fd1eebd8

$ cat example.py
from __future__ import annotations

import pytest


@pytest.mark.parametrize("a", [1], ids=[pytest.HIDDEN_PARAM])
def test_a(a: int) -> None:
    assert a == 1

$ mypy example.py 
example.py:6: error: List item 0 has incompatible type "_HiddenParam"; expected "str | float | int | None"  [list-item]
Found 1 error in 1 file (checked 1 source file)
```

--- a/changelog/14234.bugfix.rst
+++ b/changelog/14234.bugfix.rst
@@ -1,0 +1,1 @@
+Allow :ref:`pytest.HIDDEN_PARAM <hidden-param>` in :ref:`@pytest.mark.parametrize(ids=...) <pytest.mark.parametrize ref>` typing.

--- a/src/_pytest/mark/structures.py
+++ b/src/_pytest/mark/structures.py
@@ -532,7 +532,7 @@ if TYPE_CHECKING:
             argvalues: Collection[ParameterSet | Sequence[object] | object],
             *,
             indirect: bool | Sequence[str] = ...,
-            ids: Iterable[None | str | float | int | bool]
+            ids: Iterable[None | str | float | int | bool | _HiddenParam]
             | Callable[[Any], object | None]
             | None = ...,
             scope: ScopeName | None = ...,

--- a/testing/typing_checks.py
+++ b/testing/typing_checks.py
@@ -61,6 +61,12 @@ def check_testreport_attributes(report: TestReport) -> None:
     assert_type(report.location, tuple[str, int | None, str])
 
 
+# Issue #14234.
+@pytest.mark.parametrize("x", [1, 2], ids=[pytest.HIDDEN_PARAM, "visible"])
+def test_hidden_param(x: int) -> None:
+    pass
+
+
 # Test @pytest.mark.parametrize iterator argvalues deprecation.
 # Will be complain about unused type ignore if doesn't work.
 @pytest.mark.parametrize("x", iter(range(10)))  # type: ignore[deprecated]

